JetStreamCompose/jetstream/src/main/java/com/google/jetstream/presentation/screens/movies/MovieDetailsScreen.kt

```diff
@@ -52,26 +54,48 @@ object MovieDetailsScreen {
 
 @Composable
 fun MovieDetailsScreen(
-    movieId: String,
     goToMoviePlayer: () -> Unit,
     onBackPressed: () -> Unit,
-    refreshScreenWithNewMovie: (Movie) -> Unit
+    refreshScreenWithNewMovie: (Movie) -> Unit,
+    movieDetailsScreenViewModel: MovieDetailsScreenViewModel = hiltViewModel()
 ) {
-    val childPadding = rememberChildPadding()
-    val screenHeight = LocalConfiguration.current.screenHeightDp.dp
-    val movieRepository = LocalMovieRepository.current!!
-    val movieDetails = remember { movieRepository.getMovieDetails(movieId = movieId) }
+    val uiState by movieDetailsScreenViewModel.uiState.collectAsState()
```

Comment:
`collectAsStateWithLifecycle` is a safer and preferred alternative. Check out this [awesome article from Manu](https://medium.com/androiddevelopers/consuming-flows-safely-in-jetpack-compose-cde014d0d5a3)